pkg/server/internal/lib/server.go

```diff
@@ -0,0 +1,67 @@
+package lib
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/cors"
+	"github.com/openelb/openelb/pkg/server/options"
+)
+
+type server struct {
+	handler http.Handler
+	options options.Options
+}
+
+func NewHTTPServer(endpoints []Endpoints, options options.Options) *server {
+	router := chi.NewRouter()
+
+	for _, endpoint := range endpoints {
+		endpoint.Register(router)
+	}
+
+	return &server{
+		handler: cors.New(cors.Options{
+			AllowedOrigins: []string{"*"},
+			AllowedMethods: []string{"GET", "POST", "PUT", "DELETE"},
+			AllowedHeaders: []string{"Accept", "Content-Type"},
+		}).Handler(router),
+		options: options,
+	}
+}
+
+func (s *server) ListenAndServe(stopCh <-chan struct{}) error {
+	server := &http.Server{
+		Addr:    fmt.Sprintf(":%d", s.options.Port),
+		Handler: s.handler,
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		<- stopCh
+		cancel()
+	}()
+
+	errCh := make(chan error)
+	go func() {
+		err := server.ListenAndServe()
+		select {
+		case errCh <- err:
+		case <-ctx.Done():
+		}
+	}()
+
+	select {
+	case err := <-errCh:
+		return err
+	case <-ctx.Done():
```

Comment:
Do we need to handle `syscall.SIGHUP`, `syscall.SIGINT`, `syscall.SIGTERM`, `syscall.SIGQUIT`? I'm not sure